campaignion_wizard/src/ConfirmStep.php

```diff
@@ -52,15 +52,22 @@ protected function generateStatusMessages() {
    * @return array
    *   Form-API array containing the submit buttons.
    */
-  protected function generateButtons($form) {
+  protected function generateButtons($form, &$form_state) {
     unset($form['buttons']['previous']);
     $buttons = [
       '#weight' => 1000,
       '#attributes' => ['class' => ['form-submit']],
     ] + $form['buttons'];
 
+    $node = $form_state['oowizard']->node;
     $can_publish = !module_exists('change_publishing_status_permission')
-      || user_access('change publishing status');
+      || user_access('change publishing status')
+      || (!user_access('change publishing status')
```

Comment:
The checks for `!user_access(…)` are redundant.

Maybe we can improve readability when we write it like:

```php
if (module_exists('…')) {
  $buttons['return']['#access'] = user_access('…') || $node->status;  // logically equivalent to $can_publish
  $buttons['draft']['#access'] = user_access('…') || !$node->status;  // logically equivalent to $can_draft
}
```